File: leetcode-submissions/404_Sum_of_Left_Leaves.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function sumOfLeftLeaves(root: TreeNode | null): number {
    const sum : number[]= new Array(1).fill(0);
    solve(root , sum);
    return sum[0];
};
function  solve(root :TreeNode | null , sum:number[]):void{
    if(!root){
        return ;
    }
    if(root.left!==null && root.left.right===null && root.left.left===null){
        sum[0]+=root.left.val;
    }
solve(root.left, sum);
solve(root.right, sum);
}

File: leetcode-submissions/404_Sum_of_Left_Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
void sumofall(TreeNode*  root,int &sum){
    if(root==NULL){
        return;
    }
    if(root->left!=NULL and root->left->right==NULL and root->left_left==NULL){
        sum+=root->left->val;
    }
    sumofall(root->left,sum);
    sumofall(root->right,sum);
}
    int sumOfLeftLeaves(TreeNode* root) {
        if(root==NULL){
            return 0;
        }
        int sum=0;
        sumofall(root,sum);
        return sum;
    }
};